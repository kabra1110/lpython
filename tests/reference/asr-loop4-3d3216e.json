{
    "basename": "asr-loop4-3d3216e",
    "cmd": "lpython --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/loop4.py",
    "infile_hash": "660eb1a895c81609d791240171fc34b6f0fea02c43de8f86420b06b7",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-loop4-3d3216e.stdout",
    "stdout_hash": "6e32b58c4dabfc67bb3b1f8c9b091b80281d761246a1a889f1300f2b",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}