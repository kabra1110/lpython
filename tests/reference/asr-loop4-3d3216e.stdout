(TranslationUnit
    (SymbolTable
        1
        {
            __main__:
                (Module
                    (SymbolTable
                        2
                        {
                            __main____global_statements:
                                (Function
                                    (SymbolTable
                                        5
                                        {
                                            
                                        })
                                    __main____global_statements
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    [test_for]
                                    []
                                    [(SubroutineCall
                                        2 test_for
                                        ()
                                        []
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            test_for:
                                (Function
                                    (SymbolTable
                                        3
                                        {
                                            i:
                                                (Variable
                                                    3
                                                    i
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    test_for
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    []
                                    []
                                    [(DoLoop
                                        ()
                                        ((Var 3 i)
                                        (IntegerConstant 0 (Integer 4))
                                        (IntegerBinOp
                                            (IntegerConstant 10 (Integer 4))
                                            Sub
                                            (IntegerConstant 1 (Integer 4))
                                            (Integer 4)
                                            (IntegerConstant 9 (Integer 4))
                                        )
                                        (IntegerConstant 1 (Integer 4)))
                                        [(If
                                            (IntegerCompare
                                                (Var 3 i)
                                                Eq
                                                (IntegerConstant 0 (Integer 4))
                                                (Logical 4)
                                                ()
                                            )
                                            [(Cycle
                                                ()
                                            )]
                                            []
                                        )
                                        (If
                                            (IntegerCompare
                                                (Var 3 i)
                                                Gt
                                                (IntegerConstant 5 (Integer 4))
                                                (Logical 4)
                                                ()
                                            )
                                            [(Exit
                                                ()
                                            )]
                                            []
                                        )
                                        (If
                                            (IntegerCompare
                                                (Var 3 i)
                                                Eq
                                                (IntegerConstant 3 (Integer 4))
                                                (Logical 4)
                                                ()
                                            )
                                            [(Stop
                                                ()
                                            )]
                                            []
                                        )]
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                )
                        })
                    __main__
                    []
                    .false.
                    .false.
                ),
            main_program:
                (Program
                    (SymbolTable
                        6
                        {
                            __main____global_statements:
                                (ExternalSymbol
                                    6
                                    __main____global_statements
                                    2 __main____global_statements
                                    __main__
                                    []
                                    __main____global_statements
                                    Public
                                )
                        })
                    main_program
                    [__main__]
                    [(SubroutineCall
                        6 __main____global_statements
                        2 __main____global_statements
                        []
                        ()
                    )]
                )
        })
    []
)
